File: algorithms/566. Reshape the Matrix/solution.py
class Solution(object):
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        n, m = len(nums), len(nums[0])
        if n * m != r * c:
            return nums
        result = []
        count = 0
        for i in xrange(n):
            for j in xrange(m):
                if count % c == 0:
                    result.append([])
                count += 1
                result[-1].append(nums[i][j])
        return result

# 56 / 56 test cases passed.
# Status: Accepted
# Runtime: 165 ms

File: algorithms/566. Reshape the Matrix/solution.cc
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int n = nums.size(), m = nums[0].size();
        if ( n * m != r * c ) {
            return nums;
        }
        vector<vector<int>> result;
        for ( int i = 0, count = 0; i < n; ++i ) {
            for ( int j = 0; j < m; ++j ) {
                if ( count % c == 0 ) {
                    result.push_back(vector<int>());
                }
                ++count;
                result.back().push_back(nums[i][j]);
            }
        }
        return result;
    }
};

// 56 / 56 test cases passed.
// Status: Accepted
// Runtime: 39 ms
// beats 100.00 %
